addon/@ember/test-waiters/build-waiter.ts

```diff
@@ -89,7 +89,7 @@ class TestWaiterImpl<T extends object | Primitive = Token> implements TestWaiter
   private _getCompletedOperations(token: T) {
     let type = typeof token;
 
-    return token !== null || (type !== 'function' && type !== 'object')
+    return token === null || (type !== 'function' && type !== 'object')
```

Comment:
Can you make this mistake harder to make?

I'm thinking something like:

```suggestion
    let isFunction = type === 'function';
    let isObject = token !== null && type === 'object';

    return isFunction || isObject
```

Reply:
Sure, how about

```js
    let isFunction = type === 'function';
    let isObject = token !== null && type === 'object';
    let isPrimitive = !isFunction && !isObject;

    return isPrimitive ? this.completedOperationsForPrimitives : this.completedOperationsForTokens;
```

Reply:
Ya, seems great to me!

Reply:
I like.